be/src/exprs/expr.cpp

```diff
@@ -654,6 +687,72 @@ ColumnRef* Expr::get_column_ref() {
     return nullptr;
 }
 
+StatusOr<LLVMDatum> Expr::generate_ir(ExprContext* context, const llvm::Module& module, llvm::IRBuilder<>& b,
+                                      const std::vector<LLVMDatum>& datums) const {
+    if (!is_compilable()) {
+        return Status::NotSupported("JIT expr not supported");
+    }
+
+    ASSIGN_OR_RETURN(auto datum, generate_ir_impl(context, module, b, datums))
+    // Unoin null.
+    if (this->is_nullable()) {
+        // TODO(Yueyang): Check this.
+        for (auto& input : datums) {
+            datum.null_flag = b.CreateOr(datum.null_flag, input.null_flag);
+        }
+    }
+    return datum;
+}
+
+void Expr::get_uncompilable_exprs(std::vector<Expr*>& exprs) {
+    if (!this->is_compilable()) {
+        exprs.emplace_back(this);
+        return;
+    }
+    for (auto child : this->children()) {
+        child->get_uncompilable_exprs(exprs);
+    }
+}
+
+void Expr::get_jit_exprs(std::vector<Expr*>& exprs) {
+    if (!this->is_compilable()) {
+        exprs.emplace_back(this);
+        return;
+    }
+    for (auto child : this->children()) {
+        child->get_jit_exprs(exprs);
+    }
+    exprs.emplace_back(this);
+}
+
+Status Expr::replace_compilable_exprs(Expr** expr, ObjectPool* pool) {
+    if ((*expr)->should_compile()) {
+        *expr = JITExpr::create(pool, *expr);
```

Comment:
no need to compile its children?

Reply:
When the ﻿`prepare()` function is executed, its child functions will be compiled.